SMDS_3/include/CGAL/tetrahedron_soup_to_triangulation_3.h

```diff
@@ -0,0 +1,216 @@
+  *     \cgalParamDescription{each element in the range describes a surface facet using the indices of points
+  *       in `points` (indices 0 to 2), and the associated `Surface_patch_index` (index 3)}
+  *     \cgalParamType{a class model of `AssociativeContainer`
+  *                    whose key type is model of `RandomAccessContainer` containing `int`
+  *                    and mapped type is `Tr::Cell::Surface_patch_index`}
+  *     \cgalParamDefault{an empty `boost::unordered_map<std::array<int, 3>, typename Tr::Cell::Surface_patch_index>`}
+  *     \cgalParamExtra{to avoid copies of large data sets, this parameter can be passed using `std::cref`}
+  *   \cgalParamNEnd
+  *   \cgalParamNBegin{subdomain_indices}
+  *     \cgalParamDescription{each element in the range gives the
+  *       `Triangulation::Cell::Subdomain_index` corresponding to the tetrahedron (cell)
+  *        of same index in `tets`}
+  *     \cgalParamType{a class model of `RandomAccessContainer` whose value type
+  *                    is `Triangulation::Cell::Subdomain_index`}
+  *     \cgalParamDefault{each finite cell of the output triangulation is
+  *                    set to have `1` as `Subdomain_index`}
+  *     \cgalParamExtra{to avoid copies of large data sets, this parameter can be passed using `std::cref`}
+  *   \cgalParamNEnd
+  *\cond SKIP_IN_MANUAL
+  *   \cgalParamNBegin{allow_non_manifold}
+  *     \cgalParamDescription{allows the construction of a triangulation with non-manifold edges
+  *       and non manifold vertices. The triangulation is invalid if this situation is met,
+  *       so it should be used only in advanced cases, and the triangulation will be hardly usable.}
+  *     \cgalParamType{bool}
+  *     \cgalParamDefault{false}
+  *   \cgalParamNEnd
+  *\endcond
+  * \cgalNamedParamsEnd
+  *
+  * @returns the 3D triangulation built from parameters
+  *
+  * @pre `points` contains each point only once
+  * @post the output triangulation must be a triangulation of the convex hull of `points`
+  * @post `is_valid()` returns `true` for the returned triangulation
+  *
+  * @sa `CGAL::Polygon_mesh_processing::polygon_soup_to_polygon_mesh()`
+  */
+  template<typename Triangulation,
+           typename PointRange,
+           typename TetrahedronRange,
+           typename NamedParameters = parameters::Default_named_parameters>
+  Triangulation tetrahedron_soup_to_triangulation_3(const PointRange& points,
+                                                    const TetrahedronRange& tets,
+                                                    const NamedParameters& np = parameters::default_values())
+  {
+    using parameters::choose_parameter;
+    using parameters::get_parameter;
+    using parameters::get_parameter_reference;
+
+    using Default_facet_map
+      = boost::unordered_map<std::array<int, 3>,
+                             typename Triangulation::Cell::Surface_patch_index>;
+    using Facet_map_ref_type
```

Comment:
@lrineau [this comment](https://github.com/CGAL/cgal/pull/5693#discussion_r824496403) from @MaelRL has been addressed here